static/src/javascripts/projects/common/modules/commercial/obervable-store.ts

```diff
@@ -0,0 +1,25 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { getPageTargeting } from './build-page-targeting';
+
+const consentObservable = new Observable<ConsentState>((subscriber) => {
+	onConsentChange((consentState) => {
+		if (consentState.tcfv2) {
+			// For tcfv2 only, the first onConsentChange is fired before the user has
+			// interacted with the consent banner. We want to ignore this first consent consentState.
+			if (consentState.tcfv2.eventStatus !== 'cmpuishown') {
+				subscriber.next(consentState);
+			}
+		} else if (consentState.ccpa || consentState.aus) {
+			subscriber.next(consentState);
+		}
+	});
+});
+
+const pageTargetingObservable = consentObservable.pipe(
+	map((consentState) => [consentState, getPageTargeting()]),
+);
```

Comment:
Don’t you want to call rebuildPageTargeting with the consent state ? 

```suggestion
const pageTargetingObservable = consentObservable.pipe(
	map((consentState) => rebuildPageTargeting(consentState)),
);
```

Reply:
I ended up not using observables - for now.